File: app/src/main/java/com/drdisagree/iconify/config/Prefs.java
package com.drdisagree.iconify.config;

import static com.drdisagree.iconify.common.Preferences.STR_NULL;
import static com.drdisagree.iconify.common.Resources.SharedPref;

import android.content.Context;
import android.content.SharedPreferences;

import com.drdisagree.iconify.Iconify;

public class Prefs {

    static SharedPreferences pref = Iconify.getAppContext().getSharedPreferences(SharedPref, Context.MODE_PRIVATE);
    static SharedPreferences.Editor editor = pref.edit();

    // Save sharedPref config
    public static void putBoolean(String key, boolean val) {
        editor.putBoolean(key, val).apply();
    }

    public static void putInt(String key, int val) {
        editor.putInt(key, val).apply();
    }

    public static void putLong(String key, long val) {
        editor.putLong(key, val).apply();
    }

    public static void putString(String key, String val) {
        editor.putString(key, val).apply();
    }

    // Load sharedPref config
    public static boolean getBoolean(String key) {
        return pref.getBoolean(key, false);
    }

    public static boolean getBoolean(String key, Boolean defValue) {
        return pref.getBoolean(key, defValue);
    }

    public static int getInt(String key) {
        return pref.getInt(key, 0);
    }

    public static int getInt(String key, int defValue) {
        return pref.getInt(key, defValue);
    }

    public static long getLong(String key) {
        return pref.getLong(key, 0);
    }

    public static long getLong(String key, long defValue) {
        return pref.getLong(key, defValue);
    }

    public static String getString(String key) {
        return pref.getString(key, STR_NULL);
    }

    public static String getString(String key, String defValue) {
        return pref.getString(key, defValue);
    }

    // Clear specific sharedPref config
    public static void clearPref(String key) {
        editor.remove(key).apply();
    }

    // Clear all sharedPref config
    public static void clearAllPrefs() {
        editor.clear().apply();
    }
}

File: app/src/main/java/com/drdisagree/iconify/config/Prefs.kt
package com.drdisagree.iconify.config

import android.content.Context
import android.content.SharedPreferences
import com.drdisagree.iconify.Iconify.Companion.appContext
import com.drdisagree.iconify.common.Preferences.STR_NULL
import com.drdisagree.iconify.common.Resources.SHARED_PREFERENCES

@Suppress("unused")
object Prefs {

    var prefs: SharedPreferences = appContext.getSharedPreferences(SHARED_PREFERENCES, Context.MODE_PRIVATE)
    private var editor: SharedPreferences.Editor = prefs.edit()

    fun putBoolean(key: String?, `val`: Boolean) {
        editor.putBoolean(key, `val`).apply()
    }

    fun putInt(key: String?, `val`: Int) {
        editor.putInt(key, `val`).apply()
    }

    fun putLong(key: String?, `val`: Long) {
        editor.putLong(key, `val`).apply()
    }

    fun putString(key: String?, `val`: String?) {
        editor.putString(key, `val`).apply()
    }

    fun getBoolean(key: String?): Boolean {
        return prefs.getBoolean(key, false)
    }

    fun getBoolean(key: String?, defValue: Boolean?): Boolean {
        return prefs.getBoolean(key, defValue!!)
    }

    fun getInt(key: String?): Int {
        return prefs.getInt(key, 0)
    }

    fun getInt(key: String?, defValue: Int): Int {
        return prefs.getInt(key, defValue)
    }

    fun getLong(key: String?): Long {
        return prefs.getLong(key, 0)
    }

    fun getLong(key: String?, defValue: Long): Long {
        return prefs.getLong(key, defValue)
    }

    fun getString(key: String?): String? {
        return prefs.getString(key, STR_NULL)
    }

    fun getString(key: String?, defValue: String?): String? {
        return prefs.getString(key, defValue)
    }

    fun clearPref(key: String?) {
        editor.remove(key).apply()
    }

    fun clearPrefs(vararg keys: String?) {
        for (key in keys) {
            editor.remove(key).apply()
        }
    }

    fun clearAllPrefs() {
        editor.clear().apply()
    }
}
